File: server/Models/Cult.cs


public class Cult : RepoItem<int>{
  public string Name  { get; set; }
  public string CoverImg { get; set; }
  public string Bio { get; set; }
  public string LeaderId { get; set; }
  public Profile Leader { get; set; }
}

// NOTE the we will get this when we are asking for the cults I am in.
public class CultViewModel : Cult
{
   public int CultMemberId { get; set; }
}

File: client/src/models/Cult.js
import { Profile } from "./Account.js"


export class Cult{
  constructor(data){
    this.id = data.id
    this.name = data.name
    this.coverImg = data.coverImg
    this.bio = data.bio
    this.leaderId = data.leaderId
    this.leader = new Profile(data.leader)
    this.createdAt = new Date(data.createdAt)
    this.updatedAt = new Date(data.updatedAt)
  }
}

export class CultViewModel extends Cult{
  constructor(data){
    super(data)
    this.cultMemberId = data.cultMemberId
  }
}
